lib/Bio/EnsEMBL/DataCheck/Checks/MemberProductionCounts.pm

```diff
@@ -0,0 +1,219 @@
+        FROM gene_member_hom_stats 
+      WHERE collection = "$collection"
+    /;
+
+    my $sums = $helper->execute( -SQL => $sql_sums, -USE_HASHREFS => 1 );
+
+    #First check counts are > 0 for the collections from sql queries
+    my $sqlFamilies = q/
+      SELECT COUNT(*) 
+        FROM family
+    /; # To be tested only with default
+    my $sqlGeneTrees = qq/
+      SELECT COUNT(*) 
+        FROM gene_tree_root 
+      WHERE clusterset_id = "$collection"
+    /;
+    my $sqlPolyploids = q/
+      SELECT COUNT(*) 
+        FROM genome_db 
+      WHERE genome_component IS NOT NULL
+    /; # Assumes that the polyploid genomes are found in all the collections
+    my $sqlCAFETrees = qq/
+      SELECT COUNT(*) 
+        FROM gene_tree_root 
+          JOIN CAFE_gene_family 
+            ON gene_tree_root.root_id = gene_tree_root_id 
+      WHERE clusterset_id = "$collection"
+    /;
+
+    my $family_count = $helper->execute_single_result( -SQL => $sqlFamilies );
+    my $genetree_count = $helper->execute_single_result( -SQL => $sqlGeneTrees );
+    my $polyploid_count = $helper->execute_single_result( -SQL => $sqlPolyploids );
+    my $cafetrees_count = $helper->execute_single_result( -SQL => $sqlCAFETrees );
+
+    my @counts = ($family_count, $genetree_count, $cafetrees_count, $genetree_count, $genetree_count, $genetree_count, $polyploid_count);
+    # Ensure rows are non-empty for each collection where expected
+    if ( $collection eq 'default' && $dba->get_division() =~ /vertebrates/ ) {
+      my $desc_1 = "The number of entries in family are > 0 for the $collection collection";
+      cmp_ok( $family_count, ">", 0, $desc_1 );
+    }
+    my $desc_2 = "The number of entries in gene_tree_root are > 0 for the $collection collection";
+    cmp_ok( $genetree_count, ">", 0, $desc_2 );
+    if ( $dba->get_division() =~ /plants/ ) {
+      my $desc_3 = "The number of entries in genome_db for polyploids are > 0 for the $collection collection";
+      cmp_ok( $polyploid_count, ">", 0, $desc_3 );
+    }
+    if ( ( $collection eq 'default' && $dba->get_division() =~ /vertebrates/ ) || ( $dba->get_division() !~ /vertebrates/) ) {
+      my $desc_4 = "The number of entries in CAFE_gene_family are > 0 for the $collection collection";
+      cmp_ok( $cafetrees_count, ">", 0, $desc_4 );
+    }
+
+    foreach my $column ( @columns[0..1] ) {
+      next if ( $dba->get_division() !~ /vertebrates/ && $column =~ /families/ );
+      next if ( $dba->get_division() =~ /vertebrates/ && $collection !~ /default/ );
+      my $col_name = "sum_$column";
+      my $desc_5 = "The sum of entries for $column in gene_member_hom_stats > 0 for the $collection collection";
+      cmp_ok( $sums->[0]->{$col_name}, ">", 0, $desc_5 );
+    }
```

Comment:
I haven't yet reached a conclusion about this. Here is what I have in mind
1. Unroll the loop and explicitly write tests for families and gene-trees
2. Should the _families_ test be moved outside of the collection loop
3. Is this redundant with the other tests below ?
What's your opinion ?

Reply:
I am re-jigging the whole thing to address your comments, I agree with you on 1 & 2 for sure. I am still thinking about the possible redundancy :woman_shrugging: